common/djangoapps/student/views.py

```diff
@@ -1644,6 +1646,11 @@ def create_account_with_params(request, params):
     if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
         params["password"] = pipeline.make_random_password()
 
+    if active_provider_requests_data_sharing(request):
+        extra_fields['data_sharing_consent'] = 'optional'
+    if active_provider_enforces_data_sharing(request, EnterpriseCustomer.AT_LOGIN):
+        extra_fields['data_sharing_consent'] = 'required'
```

Comment:
Does the code you currently have in `user_api/views.py` not set the `optional/required` status already? If not, can it do so? Would be nice to not have to add this to this file.